src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchRequestProcessor.java

```diff
@@ -163,6 +173,60 @@ public static SearchContext buildContextForRequest(
       contextBuilder.setHighlightFetchTask(highlightFetchTask);
     }
     contextBuilder.setExtraContext(new ConcurrentHashMap<>());
+
+    if (searchRequest.getInnerHitsCount() > 0) {
+      List<InnerHitFetchTask> innerHitFetchTasks =
+          new ArrayList<>(searchRequest.getInnerHitsCount());
+      for (String innerHitName : searchRequest.getInnerHitsMap().keySet()) {
+        InnerHit innerHit = searchRequest.getInnerHitsOrThrow(innerHitName);
+        Query childQuery =
+            extractQuery(indexState, "", innerHit.getInnerQuery(), innerHit.getQueryNestedPath());
+        DocCollector innerHitCollector =
+            buildDocCollector(
```

Comment:
Creating a full `DocCollector` may be overkill. Creating a relevance/sort collector could be done similar to https://github.com/Yelp/nrtsearch/blob/master/src/main/java/com/yelp/nrtsearch/server/luceneserver/search/collectors/additional/TopHitsCollectorManager.java#L120